components/Nav/Nav.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { render, screen } from '@testing-library/react';
+import Nav from './Nav';
+
+const mockPush = jest.fn();
+const mockUsePathname = jest.fn();
+
+const mockUseAuthContext = {
+  logoutAccount: jest.fn(),
+};
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+  usePathname() {
+    return mockUsePathname();
+  },
+}));
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+describe('Nav', () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
+  });
+  it('it should be hidden when path is /register', () => {
+    mockUsePathname.mockImplementation(() => '/register');
+
+    render(<Nav />);
+
+    const navElement = screen.getByTestId('nav');
+
+    expect(navElement).toBeInTheDocument();
+    expect(navElement).toHaveClass('hidden');
+  });
+
+  it('it should be hidden when path is /login', () => {
+    mockUsePathname.mockImplementation(() => '/login');
+
+    render(<Nav />);
+
+    const navElement = screen.getByTestId('nav');
+
+    expect(navElement).toBeInTheDocument();
+    expect(navElement).toHaveClass('hidden');
+  });
+
+  it('it should be visible when logged in', () => {
```

Comment:
Q: How come we don't mock the `/weeklyPicks` route for this test?

Reply:
It didn't seem like from the `<Nav/>` component that I should test any other routes besides the ones declared which were `/login` and `/register. My assumption is that the `<Nav/>` component should show up as long as its not either of those routes. Is the `/weeklyPicks` route treated differently?  Then I figured as more routes are added they should be treated the same by the `<Nav/>` component, to be visible, unless otherwise.